Update fallback WNP [fix #14501]

craigcook opened the pull request:
## One-line summary
Removes the old `index-account.html` file and updates `index.html` to remove the send-to-device widget and show a QR code to all locales, as well as a design update and adding the app store badges. This is the default WNP shown whenever there is no custom template specified; old versions, other locales, etc.

This update began life as a proposed experiment but ultimately we decided to forego the experiment and just simplify the whole thing and ship a redesigned fallback page. Hooray!

## Issue / Bugzilla link
#14501 

## Testing
http://localhost:8000/firefox/99.0/whatsnew/ (old version, all locales)
http://localhost:8000/ja/firefox/126.0/whatsnew/ (recent version but a fallback locale)

alexgibson commented on bedrock/firefox/views.py:
```diff
@@ -579,14 +579,11 @@ def get_template_names(self):
             else:
                 template = "firefox/whatsnew/index.html"
         else:
-            if show_default_account_whatsnew(version) and ftl_file_is_active("firefox/whatsnew/whatsnew-account"):
```
**Suggestion:** if we not longer need it, let's remove the `show_default_account_whatsnew()` function?

alexgibson commented on bedrock/firefox/views.py:
```diff
@@ -403,6 +402,7 @@ class WhatsnewView(L10nTemplateView):
         "firefox/whatsnew/whatsnew-fx126beta-fr.html": ["firefox/whatsnew/whatsnew"],
         "firefox/whatsnew/whatsnew-fx126-na.html": ["firefox/whatsnew/whatsnew"],
         "firefox/whatsnew/whatsnew-fx126-eu.html": ["firefox/whatsnew/whatsnew"],
+        "firefox/whatsnew/whatsnew-fx126-mobile.html": ["firefox/whatsnew/whatsnew-account", "firefox/whatsnew/whatsnew"],
```
**Question:** is this unrelated to this PR? I don't see this template anywhere?

craigcook replied:
This began life as a proposed experiment for 126, but we decided not to do that and just make this the sole fallback so I kind of hastily moved things around and clearly missed a few spots. I should have been more careful.

alexgibson commented on bedrock/firefox/views.py:
```diff
@@ -379,7 +379,6 @@ class WhatsnewView(L10nTemplateView):
         "firefox/developer/whatsnew.html": ["firefox/developer"],
         "firefox/developer/whatsnew-mdnplus.html": ["firefox/whatsnew/whatsnew-developer-mdnplus"],
         "firefox/nightly/whatsnew.html": ["firefox/nightly/whatsnew", "firefox/whatsnew/whatsnew"],
-        "firefox/whatsnew/index-account.html": ["firefox/whatsnew/whatsnew-account", "firefox/whatsnew/whatsnew"],
```
**Question:** Can we remove `firefox/whatsnew/whatsnew-account.ftl`? And also from [pontoon.toml](https://github.com/mozilla/bedrock/blob/main/l10n/configs/pontoon.toml#L194-L196)?

alexgibson commented on bedrock/firefox/templates/firefox/whatsnew/index.html:
```diff
@@ -20,41 +23,28 @@
+    <h1 class="wnp-main-title">{{ ftl('whatsnew-s2d-download-firefox-for-your') }}</h1>
+
+    <div class="qr-wrapper">
+      <img class="qr-code" src="{{ static('img/firefox/whatsnew/qrcode-firefox-whatsnew.png') }}" width="180" height="180" alt="{{ ftl('whatsnew-s2d-qr-code-alt') }}">
```
**Issue:** This `alt` text Fluent string doesn't seem to exist?

alexgibson commented on bedrock/firefox/templates/firefox/whatsnew/index-account.html:
```diff
@@ -1,80 +0,0 @@
-
-{% block js %}
-  {{ js_bundle('fxa_product_button') }}
-  {{ js_bundle('firefox_whatsnew_account') }}
```
**Suggestion:** remove this JS bundle if it's no longer needed?

alexgibson commented on bedrock/firefox/templates/firefox/whatsnew/index.html:
```diff
@@ -20,41 +23,28 @@
+</section>
 {% endblock %}
 
 {% block js %}
```
**Suggestion:** remove `"js/base/send-to-device.es6.js"` from the `firefox_whatsnew` JS bundle?

alexgibson commented on media/img/firefox/whatsnew/whatsnew_background.svg:
```diff
@@ -1 +0,0 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1313 612.51"><defs><clipPath id="a"><path fill="none" d="M-.13.97h1313v1036H-.13z"/></clipPath><clipPath id="b"><path d="M601.6-4.12C755.81 93 815.12 298.07 734.07 453.91s-271.76 203.43-426 106.31L-898.64-199.85c-154.21-97.13-213.52-302.19-132.47-458s271.75-203.43 426-106.31z" fill="none"/></clipPath><linearGradient id="c" x1="278.53" y1="188.4 ...
```
**Issue:** this image appears to still be used in the Nightly WNP? http://localhost:8000/en-US/firefox/128.0a1/whatsnew/

alexgibson reviewed (changes requested):
A few small issues and nits, but otherwise this is good clean up and it's great to see this very old page get some attention.

Last thing: do we also still need to update/remove functional tests both for this page and the old accounts page?

- https://github.com/mozilla/bedrock/blob/main/tests/functional/firefox/whatsnew/test_whatsnew.py
- https://github.com/mozilla/bedrock/blob/main/tests/functional/firefox/whatsnew/test_whatsnew_60.py

alexgibson reviewed (changes requested):
Updates all look good! Looks like we still need to update / delete the old functional tests though.

craigcook commented:
Rebased and updated. I think we probably don't need functional tests for this page since it doesn't actually do anything but display, so seems like the basic template view tests should be enough?

alexgibson commented on tests/functional/firefox/whatsnew/test_whatsnew.py:
```diff
@@ -1,28 +0,0 @@
-
-import pytest
-
-from pages.firefox.whatsnew.whatsnew import FirefoxWhatsNewPage
```
Let's please delete the associated page objects that these test files use.

craigcook replied:
doh I thought I did... 

`/tests/pages/firefox/whatsnew/whatsnew.py`
`/tests/pages/firefox/whatsnew/whatsnew_60.py`

Are there other references I missed?

alexgibson replied:
Oh sorry, I must have blanked on those! Yes this looks good, merging!

alexgibson reviewed (approved):
r+wc
